webknossos-datastore/app/com/scalableminds/webknossos/datastore/models/datasource/DataLayer.scala

```diff
@@ -115,12 +115,14 @@ object ElementClass extends ExtendedEnumeration {
     elementClassProto match {
       case ElementClassProto.uint8  => uint8
       case ElementClassProto.uint16 => uint16
+      case ElementClassProto.uint24 => uint24
       case ElementClassProto.uint32 => uint32
       case ElementClassProto.uint64 => uint64
       case ElementClassProto.int8   => int8
       case ElementClassProto.int16  => int16
       case ElementClassProto.int32  => int32
       case ElementClassProto.int64  => int64
+      case _                        => throw new RuntimeException(s"Cannot convert ElementClassProto $elementClassProto to ElementClass")
```

Comment:
please state the protobuf unrecognized field here directly, instead of the _.

Also why is there no problem for the other direction? Shouldn’t we catch float and double?

Reply:
> Also why is there no problem for the other direction? Shouldn’t we catch float and double?

I honestly don't know. Maybe Scala is smart enough to infer that this method is never called with a type that is not in the list :thinking:?!

Reply:
I'll give this a quick check :D

Reply:
hmm ... I added
```scala
val abc = ElementClass.toProto(ElementClass.float)
println(abc)
```
at the start of a route and noticed that no compilation error occurred. => So I'd say it is saver to make `toProto` return a Box and adjust the code accordingly.

Reply:
Should be better now. `toProto` now returns a box and I adjusted the cases where it was used. Moreover I removed the `elementClassToProto` implicit.

It should now be saver to use IMO